### PR TITLE
feat: Adds schema/android implementation for Bind attributes of TabView Widget

### PR DESCRIPTION
**Pull request by viniciusguardieirozup**
feat: Adds schema/android implementation for Bind attributes of TabView Widget
#283

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/296812613febd1d1f29d9f66d4f9edaf473a828e/android/beagle/src/main/java/br/com/zup/beagle/android/utils/StyleManager.kt#L41">android/beagle/src/main/java/br/com/zup/beagle/android/utils/StyleManager.kt#L41</a> - The function applyStyleComponent appears to be too complex (15). Defined complexity threshold for methods is set to ‘15’</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_backend">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/296812613febd1d1f29d9f66d4f9edaf473a828e/android/beagle/src/main/java/br/com/zup/beagle/android/utils/StyleManager.kt#L41">android/beagle/src/main/java/br/com/zup/beagle/android/utils/StyleManager.kt#L41</a> - The function applyStyleComponent appears to be too complex (15). Defined complexity threshold for methods is set to ‘15’</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/296812613febd1d1f29d9f66d4f9edaf473a828e/android/beagle/src/main/java/br/com/zup/beagle/android/utils/StyleManager.kt#L41">android/beagle/src/main/java/br/com/zup/beagle/android/utils/StyleManager.kt#L41</a> - The function applyStyleComponent appears to be too complex (15). Defined complexity threshold for methods is set to ‘15’</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/TabView.kt [hunk 1]
```diff
@@ -40,7 +42,7 @@ internal var styleManagerFactory = StyleManager()
 
 data class TabView(
     val children: List<TabItem>,
-    val styleId: String? = null
+    val styleId: Bind<String>? = null
 ) : WidgetView() {
```
Comment (LG95) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/TabView.kt:47]: Please add the non `Bind` secondary constructor.

File: backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/ui/TabView.kt [hunk 2]
```diff
@@ -28,7 +29,7 @@ import br.com.zup.beagle.core.ServerDrivenComponent
  */
 data class TabView(
     val children: List<TabItem>,
-    val styleId: String? = null
+    val styleId: Bind<String>? = null
 ) : ServerDrivenComponent
```
Comment (LG95) [inline backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/ui/TabView.kt:33]: Please add the non `Bind` secondary constructor.

File: android/sample/src/main/java/br/com/zup/beagle/sample/fragment/TabViewFragment.kt [hunk 3]
```diff
@@ -45,7 +46,7 @@ class TabViewFragment : Fragment() {
         savedInstanceState: Bundle?
     ): View? {
         val declarative = TabView(
-            styleId = "DesignSystem.TabView.Custom",
```
Comment (LG95) [inline android/sample/src/main/java/br/com/zup/beagle/sample/fragment/TabViewFragment.kt:48]: Please use the non `Bind` secondary constructor.

File: android/beagle/src/test/java/br/com/zup/beagle/android/data/serializer/BeagleMoshiTest.kt [hunk 4]
```diff
@@ -290,9 +291,9 @@ class BeagleMoshiTest: BaseTest() {
     }
 
     @Test
-    fun make_should_return_moshi_to_serialize_a_TabView(){
+    fun make_should_return_moshi_to_serialize_a_TabView() {
         // Given
-        val component = TabView(children = listOf())
+        val component = TabView(children = listOf<TabItem>())
```
Comment (LG95) [inline android/beagle/src/test/java/br/com/zup/beagle/android/data/serializer/BeagleMoshiTest.kt:296]: Why is this now necessary?

File: android/sample/src/main/java/br/com/zup/beagle/sample/fragment/TabViewFragment.kt [hunk 5]
```diff
@@ -45,7 +46,7 @@ class TabViewFragment : Fragment() {
...
+            styleId = Bind.valueOf("DesignSystem.TabView.Custom"),
```
Comment (LG95) [inline android/sample/src/main/java/br/com/zup/beagle/sample/fragment/TabViewFragment.kt:49]:
```suggestion
            styleId = "DesignSystem.TabView.Custom",
```

**Pull request merged**